NetSerializer/Primitives.cs

```diff
@@ -255,16 +256,146 @@ public static void ReadPrimitive(Stream stream, out float value)
 		public static void WritePrimitive(Stream stream, double value)
 		{
 			ulong v = (ulong)BitConverter.DoubleToInt64Bits(value);
-			WriteVarint64(stream, v);
+			WriteUInt64(stream, v);
 		}
 
 		public static void ReadPrimitive(Stream stream, out double value)
 		{
-			ulong v = ReadVarint64(stream);
+			ulong v = ReadUInt64(stream);
 			value = BitConverter.Int64BitsToDouble((long)v);
 		}
 #endif
 
+		private static void WriteUInt16(Stream stream, ushort value)
+	    {
+	        stream.WriteByte((byte) value);
+	        stream.WriteByte((byte) (value >> 8));
+	    }
+
+		private static ushort ReadUInt16(Stream stream)
+		{
+            ushort a = 0;
+
+            for (var i = 0; i < 16; i += 8)
+            {
+                var val = stream.ReadByte();
+                if (val == -1)
+                    throw new EndOfStreamException();
+
+                a |= (ushort) (val << i);
+            }
+
+            return a;
+		}
+
+		// 32 and 64 bit variants use stackalloc when everything is available since it's faster.
+
+#if !NETCOREAPP
+		private static void WriteUInt32(Stream stream, uint value)
+		{
+			stream.WriteByte((byte) value);
+			stream.WriteByte((byte) (value >> 8));
+			stream.WriteByte((byte) (value >> 16));
+			stream.WriteByte((byte) (value >> 24));
+		}
+
+		private static void WriteUInt64(Stream stream, ulong value)
+		{
+			stream.WriteByte((byte) value);
+			stream.WriteByte((byte) (value >> 8));
+			stream.WriteByte((byte) (value >> 16));
+			stream.WriteByte((byte) (value >> 24));
+			stream.WriteByte((byte) (value >> 32));
+			stream.WriteByte((byte) (value >> 40));
+			stream.WriteByte((byte) (value >> 48));
+			stream.WriteByte((byte) (value >> 56));
+		}
+
+		private static uint ReadUInt32(Stream stream)
+		{
+			uint a = 0;
+
+			for (var i = 0; i < 32; i += 8)
+			{
+				var val = stream.ReadByte();
+				if (val < 0)
+					throw new EndOfStreamException();
+
+				a |= (uint)val << i;
+			}
+
+			return a;
+		}
+
+        private static ulong ReadUInt64(Stream stream)
+        {
+	        ulong a = 0;
+
+            for (var i = 0; i < 64; i += 8)
+            {
+            	var val = stream.ReadByte();
+            	if (val < 0)
+            		throw new EndOfStreamException();
+
+            	a |= (ulong)val << i;
+            }
```

Comment:
Indentation here is done with spaces instead of tabs. Github doesn't seem to show that, but "git show" shows it as red.

Reply:
No idea how Rider messed this one up. Fixed now.